src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -71,62 +72,60 @@ private function stream(): void
         $generators = [];
         $structure = $this->data;
 
-        array_walk_recursive($structure, function (&$item, $key) use (&$generators) {
+        array_walk_recursive($structure, function (&$item) use (&$generators) {
             // generators should be used but for better DX all kind of Traversable are supported
             if ($item instanceof \Traversable && !$item instanceof \JsonSerializable) {
-                // using uniqid to avoid conflict with eventually other data in the structure
-                $placeholder = uniqid('__placeholder_', true);
-                $generators[$placeholder] = $item;
-
-                $item = $placeholder;
+                $generators[] = $item;
+                $item = self::PLACEHOLDER;
+            } elseif (self::PLACEHOLDER === $item) {
+                $generators[] = $item;
             }
         });
 
         $jsonEncodingOptions = \JSON_THROW_ON_ERROR | $this->getEncodingOptions();
-        $structureText = json_encode($structure, $jsonEncodingOptions);
-
-        foreach (array_keys($generators) as $placeholder) {
-            // split structure json by placeholder for stream the before and between part of the generator items
-            [$start, $end] = explode('"'.$placeholder.'"', $structureText, 2);
-
-            // send first and between parts of the structure
-            echo $start;
-
-            $count = 0;
-            $startTag = '[';
-            foreach ($generators[$placeholder] as $key => $item) {
-                if (0 === $count) {
-                    // depending of the first elements key the generator is detected as a list or map
-                    // we can not check for a whole list or map because that would hurt the performance
-                    // of the streamed response which is the main goal of this response class
-                    if ($key !== 0) {
-                        $startTag = '{';
-                    }
+        $jsonParts = explode('"'.self::PLACEHOLDER.'"', json_encode($structure, $jsonEncodingOptions));
+        foreach ($jsonParts as $index => $jsonPart) {
+            echo $jsonPart;
 
-                    echo $startTag;
-                } else {
-                    // if not first element of the generic a separator is required between the elements
-                    echo ',';
-                }
+            if (!isset($generators[$index])) {
+                break;
+            }
```

Comment:
This should never happen and is mostly an error in the code.

If we want to check for it we should do:

```suggestion
            assert(!isset($generators[$index]), \'Expected Generator not found.\');
```

Reply:
It always happen with the last json part. There is `count($jsonParts) === count($generator) + 1`

Reply:
I updated to loop over `$generator` and remove this hack.